tests/apollo/util/bft.py

```diff
@@ -1162,6 +1162,29 @@ async def force_quorum_including_replica(self, replica_id):
             self.stop_replicas(self.random_set_of_replicas(
                 len(self.procs) - (2 * self.config.f + self.config.c + 1), without={primary, replica_id}))
 
+
+    async def wait_for_getting_missing_blocks_state(self, replica_id):
```

Comment:
What if cycle is ended already?
this is an issue in all out tests I think.
I think that the idea is to have such a long cycle to make the chance for the cycle to end when entering this function == 0.
One of the way to be sure about this is convert cycleCounter_ to a metric called something like "current_cycle_id" and have another metric called last_cycle_id.
Id  current_cycle_id is 1 and last_cycle_id is 0 -> means we are still in the middle of cycle 1. if both are equal - it means that cycle is done.